src/XR/features/WebXRNearInteraction.ts

```diff
@@ -321,148 +361,172 @@ export class WebXRNearInteraction extends WebXRAbstractFeature {
                 }
                 return pick;
             };
-            let populateNearInteractionInfo = (nearInteractionInfo: Nullable<PickingInfo>): Nullable<PickingInfo> => {
-                let result = null;
+            let populateNearInteractionInfo = (nearInteractionInfo: Nullable<PickingInfo>): PickingInfo => {
+                let result = new PickingInfo();
+
                 let nearInteractionAtOrigin = false;
                 let nearInteraction = nearInteractionInfo && nearInteractionInfo.pickedPoint && nearInteractionInfo.hit;
                 if (nearInteractionInfo?.pickedPoint) {
                     nearInteractionAtOrigin = nearInteractionInfo.pickedPoint.x === 0 && nearInteractionInfo.pickedPoint.y === 0 && nearInteractionInfo.pickedPoint.z === 0;
                 }
                 if (nearInteraction && !nearInteractionAtOrigin) {
-                    result = nearInteractionInfo;
-                    controllerData.nearInteraction = true;
+                    result = nearInteractionInfo!;
                 }
                 return result;
             };
 
-            let pick = null;
-            let hoverRange = false;
+            let pick = new PickingInfo();

```

Comment:
there is no reason to init pick here. It is populated (or replaced) in the next few loops. One less object to garbage-collect on each frame.